nannou/src/wgpu/texture/mod.rs

```diff
@@ -35,9 +35,9 @@ pub struct Texture {
 /// be of the whole texture, but it might also be of some sub-section of the texture. When an API
 /// provides
 #[derive(Debug)]
-pub struct TextureView {
+pub struct TextureView<'t> {
     handle: Arc<TextureViewHandle>,
-    descriptor: wgpu::TextureViewDescriptor,
+    descriptor: wgpu::TextureViewDescriptor<'t>,
```

Comment:
Hey @alekspickle, it looks like you might be running into some issues with Rust's lifetimes, and my suspicion is that this field here is the source of the unexpected complexity :) 

The lifetime parameter on the `wgpu::TextureViewDescriptor` is associated with the lifetime of its `label` field. The purpose of this `label` field is to make it easier to track `TextureView`s when debugging or using some inspector like renderdoc. By remaining generic over the lifetime, wgpu allows users to use a label that is a reference to some static slice of memory (e.g. string literals like "foo", `const` or `static` `str`s) or dynamically allocated one (e.g. a reference to a `String`).

Throughout nannou, the only descriptor labels we use are string literals (e.g. "frame-texture-vew") as these are almost always enough to quickly identify the resource in question when using a graphics debugger / inspector. As a result, our `wgpu::TextureViewDescriptor`'s always have `'static` lifetimes. In case you're unfamiliar, `'static` lifetimes are special in that they indicate a lifetime is valid for the entire duration the program will run.

We should be able to get away without `TextureView` having a lifetime parameter, and instead declaring the `descriptor` field like this:

```rust
pub struct TextureView {
    handle: Arc<TextureViewHandle>,
    descriptor: wgpu::TextureViewDescriptor<'static>, // notice the 'static lifetime here
    texture_extent: wgpu::Extent3d,
    texture_id: TextureId,
}
```

This should hopefully allow you to avoid this problem where the lifetime parameters seem to propagate up through the whole program and it becomes impossible to make the borrow checker happy.

I hope this makes sense / helps a bit!

Reply:
Indeed, that could simplify things. 
But I guess the reason is me understanding wgpu/nannou code intricacies enough.